core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/environment/EnvironmentV4Endpoint.java

```diff
@@ -41,75 +41,67 @@
     @ApiOperation(value = EnvironmentOpDescription.CREATE, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "createEnvironment")
-    DetailedEnvironmentV4Response post(@PathParam("workspaceId") Long workspaceId, @Valid EnvironmentV4Request request);
+    DetailedEnvironmentResponse post(@PathParam("workspaceId") Long workspaceId, @Valid EnvironmentRequest request);
 
     @GET
-    @Path("/{name}")
+    @Path("/{crn}")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.GET, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "getEnvironment")
-    DetailedEnvironmentV4Response get(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName);
+    DetailedEnvironmentResponse get(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn);
 
     @DELETE
-    @Path("/{name}")
+    @Path("/{crn}")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.DELETE, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "deleteEnvironment")
-    SimpleEnvironmentV4Response delete(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName);
+    SimpleEnvironmentResponse delete(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn);
 
     @PUT
-    @Path("/{name}")
+    @Path("/{crn}")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.EDIT, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "editEnvironment")
-    DetailedEnvironmentV4Response edit(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
-            @NotNull EnvironmentEditV4Request request);
+    DetailedEnvironmentResponse edit(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
+            @NotNull EnvironmentEditRequest request);
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.LIST, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "listEnvironment")
-    SimpleEnvironmentV4Responses list(@PathParam("workspaceId") Long workspaceId);
+    SimpleEnvironmentResponses list(@PathParam("workspaceId") Long workspaceId);
 
     @PUT
-    @Path("/{name}/attach")
+    @Path("/{crn}/attach")
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = EnvironmentOpDescription.ATTACH_RESOURCES, produces = ContentType.JSON, notes = Notes.ENVIRONMENT_NOTES,
             nickname = "attachResourcesToEnvironment")
-    DetailedEnvironmentV4Response attach(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String environmentName,
+    DetailedEnvironmentV4Response attach(@PathParam("workspaceId") Long workspaceId, @PathParam("crn") String environmentCrn,
```

Comment:
Shouldn't it be DetailedEnvironmentResponse ?

Reply:
Nope, this is the old Endpoint in cloudbreak which I left here to make it easier to rewire integration tests to use the new env service. If I had deleted this class, most of the integration tests should be deleted too. And it wouldn't be much fun rewriting them.